mvj/settings.py

```diff
@@ -132,6 +132,13 @@
         'django_filters.rest_framework.DjangoFilterBackend',
     ],
     'DEFAULT_METADATA_CLASS': 'leasing.metadata.FieldsMetadata',
+    'DEFAULT_RENDERER_CLASSES': [
+        'rest_framework.renderers.JSONRenderer',
+        # Render the browsable API without the html form because the form uses RelatedField.get_choices. Which
+        # needs that serializers to_representation returns a hashable value. We return the whole instance
+        # which is not hashable.
+        'leasing.renderers.BrowsableAPIRendererWithoutHtmlForm',
+    ],
```

Comment:
I think you should rather override the get_choices method in InstanceDictPrimaryKeyRelatedField.

Let's see.. maybe this could work?
```python
    def get_choices(self, cutoff=None):
        queryset = self.get_queryset()
        if queryset is None:
            return {}
        if cutoff is not None:
            queryset = queryset[:cutoff]
        return OrderedDict(
            (item.pk, self.display_value(item))
            for item in queryset)
```